src/components/Admin/Admin.js

```diff
@@ -25,7 +25,7 @@ const AdminPage = props => {
 
       return [setLoading(false), unsubscribe()];
     });
-  }, [users]);
+  }, []);
```

Comment:
I don't know why there should be an array parameter for useEffect hook yet. But I think the memory leak is resulted from your recursive function `unsubscribe`.

Reply:
@bt-s

Reply:
Maybe you meant to write something like this:
`
useEffect(() => {
    setLoading(true);

    const unsubscribe = props.firebase.users().onSnapshot(snapshot => {
      let users = [];

      snapshot.forEach(doc => users.push({...doc.data(), uid: doc.id}));

      setUsers(users);
      setLoading(false);
}   // **this closing bracket should above return? **
      return [setLoading(false), unsubscribe()];
    );
`

Reply:
> I don't know why there should be an array parameter for useEffect hook yet. But I think the memory leak is resulted from your recursive function `unsubscribe`.

I think I'm wrong about that recursive function. Currently it should never be called since it wasn't returned outside the scope of anonymous function passed to useEffect(). So it has nothing to do with the memory leak issue. Also, I realized that I'm not sure what you mean by saying memory leak. 

Another thing I know just now is that passing an empty array to useEffect() was to make sure that callback function was called only once.